device-connectors/src/testflinger_device_connectors/devices/maas2/maas2.py

```diff
@@ -70,19 +71,22 @@ def recover(self):
         self.node_release()
 
     def provision(self):
-        if self.config.get("reset_efi"):
-            self.reset_efi()
-        # Check if this is a device where we need to clear the tpm (dawson)
-        if self.config.get("clear_tpm"):
-            self.clear_tpm()
-        provision_data = self.job_data.get("provision_data")
-        # Default to a safe LTS if no distro is specified
-        distro = provision_data.get("distro", "xenial")
-        kernel = provision_data.get("kernel")
-        user_data = provision_data.get("user_data")
-        storage_data = provision_data.get("disks")
-
-        self.deploy_node(distro, kernel, user_data, storage_data)
+        try:
+            if self.config.get("reset_efi"):
+                self.reset_efi()
+            # Check if this is a device where we need to clear the tpm (dawson)
+            if self.config.get("clear_tpm"):
+                self.clear_tpm()
+            provision_data = self.job_data.get("provision_data")
+            # Default to a safe LTS if no distro is specified
+            distro = provision_data.get("distro", "xenial")
+            kernel = provision_data.get("kernel")
+            user_data = provision_data.get("user_data")
+            storage_data = provision_data.get("disks")
+            self.deploy_node(distro, kernel, user_data, storage_data)
+        except Exception as err:
```

Comment:
I wonder if we could catch this at an even higher level? We'd like to do something similar for other device connectors eventually, so it would be really nice if we didn't have to repeat this part of the work. If it fails, then we may have some steps here where we *want* to catch the exception, do some cleanup, and re-raise it. But if we catch it up at the point where we run the subcommand, or the command as a whole, then we can catch it for everything - possibly even errors that we haven't thought of.